thriftpy2/parser/parser.py

```diff
@@ -26,10 +26,12 @@
 
 
 def p_error(p):
+    thrift = threadlocal.thrift_stack[-1]
     if p is None:
-        raise ThriftGrammarError('Grammar error at EOF')
-    raise ThriftGrammarError('Grammar error %r at line %d' %
-                             (p.value, p.lineno))
+        raise ThriftGrammarError('Grammar error at EOF of file path in %s' % thrift.__thrift_file__)
+
+    raise ThriftGrammarError('Grammar error %r at line %d of file path in %s, ' %
```

Comment:
```suggestion
        raise ThriftGrammarError("Grammar error at EOF of the file '%s'" % thrift.__thrift_file__)

    raise ThriftGrammarError("Grammar error %r at line %d of the file '%s'" %
```

This will make the error message more concise.